### PR TITLE
update armature

### PR DESCRIPTION
**Pull request by 2youyou2**
1.add support for tweenRotate property
2.change const char\* to const std::string&

### CODE REVIEW

File: cocos/editor-support/cocostudio/CCArmature.cpp [hunk 1]
```diff
@@ -233,12 +231,12 @@ Bone *Armature::createBone(const char *boneName)
...
 
 
-void Armature::addBone(Bone *bone, const char *parentName)
+void Armature::addBone(Bone *bone, const std::string& parentName)
 {
     CCASSERT( bone != nullptr, "Argument must be non-nil");
     CCASSERT(_boneDic.at(bone->getName()) == nullptr, "bone already added. It can't be added again");
 
-    if (nullptr != parentName)
+    if ("" != parentName)
```
Comment (dumganhar) [inline cocos/editor-support/cocostudio/CCArmature.cpp]: Using `!parentName.empty()` will be better.

File: cocos/editor-support/cocostudio/CCArmature.cpp [hunk 2]
```diff
@@ -294,7 +292,7 @@ void Armature::changeBoneParent(Bone *bone, const char *parentName)
         bone->setParentBone(nullptr);
     }
 
-    if (parentName != nullptr)
+    if (parentName != "")
```
Comment (dumganhar) [inline cocos/editor-support/cocostudio/CCArmature.cpp]: The same here.

File: cocos/editor-support/cocostudio/CCDataReaderHelper.cpp [hunk 3]
```diff
@@ -1184,7 +1180,7 @@ ContourData *DataReaderHelper::decodeContour(tinyxml2::XMLElement *contourXML, D
 
 
 
-void DataReaderHelper::addDataFromJsonCache(const char *fileContent, DataInfo *dataInfo)
+void DataReaderHelper::addDataFromJsonCache(const char* fileContent, DataInfo *dataInfo)
```
Comment (dumganhar) [inline cocos/editor-support/cocostudio/CCDataReaderHelper.cpp]: Why is it a `const std::string&` type here?

Comment (2youyou2) [inline cocos/editor-support/cocostudio/CCDataReaderHelper.cpp]: I use it here :   

char *pFileContent = (char *)CCFileUtils::getInstance()->getFileData(fullPath.c_str() , "r", &size);
DataReaderHelper::addDataFromCache(pFileContent, &dataInfo);

Using const std::string& will create a new string and copy the file content。
Do i need to do this?

Comment (dumganhar) [inline cocos/editor-support/cocostudio/CCDataReaderHelper.cpp]: Please be careful of dealing with `FileUtils::getFileData`, the return value is not `\0` terminal string.
If you expect it's a string, you need to `malloc` new memory of which size is `size+1`, and set the last element to `'\0'`.

```
char* myContent = (char*)malloc((size+1) * sizeof(char));
myContent[size] = '\0';
memcpy(myContent, pFileContent, size * sizeof(char));

...

free(myContent);

```

Comment (dumganhar) [inline cocos/editor-support/cocostudio/CCDataReaderHelper.cpp]: Also notice you need to `free` the memory `pFileContent` points to .
While on `master` branch, you need to use `delete[]`.

File: samples/Cpp/TestCpp/Classes/ExtensionsTest/CocoStudioArmatureTest/ArmatureScene.cpp [hunk 4]
```diff
@@ -535,20 +535,18 @@ void TestAnimationEvent::onEnter()
...
 }
-void TestAnimationEvent::animationEvent(Armature *armature, MovementEventType movementType, const char *movementID)
+void TestAnimationEvent::animationEvent(Armature *armature, MovementEventType movementType, const std::string& movementID)
 {
-    std::string id = movementID;
-
     if (movementType == LOOP_COMPLETE)
     {
-        if (id.compare("Fire") == 0)
+        if (movementID.compare("Fire") == 0)
```
Comment (dumganhar) [inline samples/Cpp/TestCpp/Classes/ExtensionsTest/CocoStudioArmatureTest/ArmatureScene.cpp]: Using `==` operator will be better.

File: samples/Cpp/TestCpp/Classes/ExtensionsTest/CocoStudioArmatureTest/ArmatureScene.cpp [hunk 5]
```diff
@@ -535,20 +535,18 @@ void TestAnimationEvent::onEnter()
...
         {
             ActionInterval *actionToRight = MoveTo::create(2, Point(VisibleRect::right().x - 50, VisibleRect::right().y));
             armature->stopAllActions();
             armature->runAction(Sequence::create(actionToRight,  CallFunc::create( CC_CALLBACK_0(TestAnimationEvent::callback1, this)), nullptr));
             armature->getAnimation()->play("Walk");
         }
-        else if (id.compare("FireMax") == 0)
+        else if (movementID.compare("FireMax") == 0)
```
Comment (dumganhar) [inline samples/Cpp/TestCpp/Classes/ExtensionsTest/CocoStudioArmatureTest/ArmatureScene.cpp]: The same here.

**Comment by dumganhar** [comment]
Great, thanks a lot. :)

**Pull request merged**